cmd/db_opener_sequel_pro.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"fmt"
+
+	"io/ioutil"
+	"text/template"
+
+	"github.com/mitchellh/cli"
+)
+
+type DBOpenerSequelPro struct {
+	ui cli.Ui
+}
+
+var sequelProSpfTemplate = `
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+  <key>ContentFilters</key>
+  <dict/>
+  <key>auto_connect</key>
+  <true/>
+  <key>data</key>
+  <dict>
+    <key>connection</key>
+    <dict>
+      <key>database</key>
+      <string>{{.DBName}}</string>
+      <key>host</key>
+      <string>{{.DBHost}}</string>
+      <key>user</key>
+      <string>{{.DBUser}}</string>
+      <key>password</key>
+      <string>{{.DBPassword}}</string>
+      <key>ssh_host</key>
+      <string>{{.SSHHost}}</string>
+      <key>ssh_port</key>
+      <string>{{.SSHPort}}</string>
+      <key>ssh_user</key>
+      <string>{{.SSHUser}}</string>
+      <key>type</key>
+      <string>SPSSHTunnelConnection</string>
+    </dict>
+  </dict>
+  <key>format</key>
+  <string>connection</string>
+  <key>queryFavorites</key>
+  <array/>
+  <key>queryHistory</key>
+  <array/>
+</dict>
+</plist>
+`
+
+func (o *DBOpenerSequelPro) open(c DBCredentials) (err error) {
+	sequelProSpf, sequelProSpfErr := ioutil.TempFile("", "*.spf")
+	if sequelProSpfErr != nil {
+		return fmt.Errorf("Error creating temporary SequelPro SPF file: %s", sequelProSpfErr)
+	}
+	// TODO: [Help Wanted] There is a chance that the SPF file got deleted before SequelPro establish db connection.
```

Comment:
Can you explain this more? Was it getting deleted by the `defer` first?

Reply:
> Was it getting deleted by the defer first?

That is my theory.

If we uncomment the `defer` line, sequel pro will open but doesn't pre-fill any db info.
If we comment the `defer` line, sequel pro will open with db info pre-filled and try to connect to the db.

I suspect sequel pro needs to read that spf file after launch, but golang deleted the spf too soon.

Reply:
Makes some sense. You could verify by logging like this:

```go
func (o *DBOpenerSequelPro) open(c DBCredentials) (err error) {
         # etc
	open := execCommand("open", sequelProSpf.Name())
	err := cmd.Start()
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("running sequel pro...")
	err = cmd.Wait()
	log.Printf("Command finished with error: %v", err)
}
```

And then add a log printf to `deleteFile` to see when it's actually run compared to the other log lines

Reply:
Confirmed my theroy - the SPF file is read twice:
- once by OS to open sequel pro
- once by sequel pro to get db credentials

---

Newer version of sequel pro shows this error message if SPF is deleted too early:

```
Error while reading connection data file

Connection data file couldn't be read. (The file “804911772.spf” couldn’t be opened because there is no such file.)
```